overlord/snapstate/snapmgr.go

```diff
@@ -289,7 +289,18 @@ func (m *SnapManager) doClearSnapData(t *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	return m.backend.RemoveSnapData(info)
+	if err = m.backend.RemoveSnapData(info); err != nil {
+		return err
+	}
+
+	// Only remove data shared between versions if this is the last version
```

Comment:
This is a bit problematic, and makes me want to think further about the whole approach we're taking. Imagine a button that says "Reset snap data".. if we keep the shared data around we're not actually resetting the data. We'll restart the snap, and its data is still there, partially! How do we reset the data for a snap, ever, once we have a shared directory?